### PR TITLE
WIP:Webpack4

### PR DESCRIPTION
**Pull request by codeBelt**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,87 +52,84 @@
...
-    "@types/classnames": "2.2.3",
-    "@types/fs-extra": "5.0.2",
-    "@types/hapi": "17.0.10",
-    "@types/history": "4.6.2",
-    "@types/inert": "5.1.0",
-    "@types/node": "9.6.5",
+    "@babel/cli": "7.1.5",
+    "@babel/core": "7.1.5",
+    "@babel/plugin-proposal-class-properties": "7.1.0",
+    "@babel/plugin-proposal-object-rest-spread": "7.0.0",
```
Comment (n8rzz) [inline package.json]: do you still need this?  even if babel needs it, isn't this covered by `tsc`?

Comment (codeBelt) [reply package.json]: Not 100% but I am using Babel 7 to compile the `ts` now and not `tsc`

Comment (n8rzz) [reply package.json]: Let's talk about that tomorrow.  Why would you use an abstraction of ts to compile ts?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "typescript-hapi-react-hot-loader-example",
-  "version": "1.0.0",
+  "version": "2.0.0",
   "description": "Simple React Hot Loading example with Hapi Server-side rendering",
   "engines": {
     "node": ">=8.9.1"
```
Comment (ccheney) [inline package.json]:
```suggestion
     "node": ">=10.11.0"
```

File: src/server/ServerManager.ts [hunk 3]
```diff
@@ -37,6 +37,11 @@ class ServerManager {
     }
 
     public async startServer(): Promise<void> {
+        process.on('unhandledRejection', (err) => {
```
Comment (ccheney) [inline src/server/ServerManager.ts]: missing arg type?

```suggestion
        process.on('unhandledRejection', (err: Error) => {
```

File: src/server/ServerManager.ts [hunk 4]
```diff
@@ -37,6 +37,11 @@ class ServerManager {
...
+            console.log(err);
```
Comment (ccheney) [inline src/server/ServerManager.ts]:
```suggestion
            console.error(err);
```

File: src/utilities/ProviderUtility.ts [hunk 5]
```diff
@@ -5,12 +5,11 @@ import rootReducer from '../stores/rootReducer';
 import {composeWithDevTools} from 'redux-devtools-extension/developmentOnly';
 import createSagaMiddleware, {END, SagaMiddleware} from 'redux-saga';
 import rootSaga from '../stores/rootSaga';
-import IStore from '../stores/IStore';
 import ISagaStore from '../stores/ISagaStore';
 
 class ProviderUtility {
 
-    public static createProviderStore(initialState: any = {}, history: History = null, isServerSide: boolean = false): ISagaStore<IStore> {
+    public static createProviderStore(initialState: any = {}, history: History = null, isServerSide: boolean = false): ISagaStore {
```
Comment (ccheney) [inline src/utilities/ProviderUtility.ts]:
```suggestion
    public static createProviderStore(initialState: object = {}, history: History = null, isServerSide: boolean = false): ISagaStore {
```

**Pull request closed without merging**